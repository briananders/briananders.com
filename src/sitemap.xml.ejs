<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"
        xmlns:image="http://www.google.com/schemas/sitemap-image/1.1"
        xmlns:mobile="http://www.google.com/schemas/sitemap-mobile/1.0"
        xmlns:news="http://www.google.com/schemas/sitemap-news/0.9"
        xmlns:xhtml="http://www.w3.org/1999/xhtml"
  ><%
    for(let i = 0; i < locals.pages.length; i++) {
      const page = locals.pages[i];
      const date = page.data.date || new Date();
      const dateOutput = `${date.getFullYear()}-${('00' + (date.getMonth() + 1)).slice(-2)}-${('00' + date.getDate()).slice(-2)}`;
      if(page.data.robots !== false) {
  %>
  <url>
    <loc><%- locals.siteData.domain %><%- page.url %></loc>
    <lastmod><%- dateOutput %></lastmod>
    <priority><%- page.data.priority %></priority>
    <changefreq>yearly</changefreq>
  </url><%
      }
    }
  %>
</urlset>