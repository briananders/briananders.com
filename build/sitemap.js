const ejs = require('ejs');
const fs = require('fs-extra');

module.exports = function sitemap(dir, completionFlags, buildEvents, pageMappingData) {
  const timestamp = require(`${dir.build}timestamp`);
  const siteData = require(`${dir.build}site-data`)(dir);

  console.log(`${timestamp.stamp()}: sitemap()`);
  ejs.renderFile(`${dir.src}sitemap.xml.ejs`, {
    pages: pageMappingData,
    siteData,
  }, {
    compileDebug: true,
  }, (error, str) => {
    if (error) throw error;
    fs.writeFile(`${dir.package}sitemap.xml`, str, (err) => {
      if (err) throw err;

      completionFlags.SITE_MAP = true;
      buildEvents.emit('sitemap-done');
    });
  });
};
