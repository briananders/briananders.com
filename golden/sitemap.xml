<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="https://www.sitemaps.org/schemas/sitemap/0.9"
        xmlns:image="https://www.google.com/schemas/sitemap-image/1.1"
        xmlns:mobile="https://www.google.com/schemas/sitemap-mobile/1.0"
        xmlns:news="https://www.google.com/schemas/sitemap-news/0.9"
        xmlns:xhtml="https://www.w3.org/1999/xhtml"
  >
  <url>
    <loc>https://briananders.com/about/</loc>
    <lastmod>2024-08-20</lastmod>
    <priority>0.9</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/drums/</loc>
    <lastmod>2022-02-17</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/</loc>
    <lastmod>2025-10-19</lastmod>
    <priority>1</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/2d-automaton/</loc>
    <lastmod>2018-03-03</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/ant-life-simulator/</loc>
    <lastmod>2022-05-22</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/blue-green/</loc>
    <lastmod>2022-01-13</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/browser-and-device-diagnostics/</loc>
    <lastmod>2023-03-24</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/canvas-static/</loc>
    <lastmod>2023-09-03</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/cellular-automaton/</loc>
    <lastmod>2018-02-22</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/circle-multiplier/</loc>
    <lastmod>2022-03-27</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/circle-wave-illusion/</loc>
    <lastmod>2022-06-25</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/color-canvas/</loc>
    <lastmod>2018-08-01</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/design-system/</loc>
    <lastmod>2024-12-30</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/earth-rotating-sprite-animation/</loc>
    <lastmod>2018-08-02</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/how-to-get-rid-of-untracked-files-from-your-git-branch/</loc>
    <lastmod>2018-11-02</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/how-to-git-merge-without-being-prompted-for-a-message/</loc>
    <lastmod>2018-11-02</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/how-to-match-the-height-of-sibling-elements/</loc>
    <lastmod>2016-11-16</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/in-view-text-animation/</loc>
    <lastmod>2020-10-17</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/</loc>
    <lastmod>2016-08-31</lastmod>
    <priority>0.9</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/interview-question/</loc>
    <lastmod>2020-05-05</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/last-fm-scrobbles/</loc>
    <lastmod>2025-01-01</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/last-fm/</loc>
    <lastmod>2017-09-06</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/line-circle-rotator/</loc>
    <lastmod>2022-01-15</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/lissajous-curve/</loc>
    <lastmod>2019-02-14</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/making-animations-with-squares/</loc>
    <lastmod>2022-03-25</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/minesweeper/</loc>
    <lastmod>2016-09-09</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/moire-pattern-colors/</loc>
    <lastmod>2022-06-23</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/moire-patterns/</loc>
    <lastmod>2022-06-22</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/my-favorite-aliases/</loc>
    <lastmod>2018-11-03</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/my-favorite-git-aliases/</loc>
    <lastmod>2018-11-03</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/my-git-config/</loc>
    <lastmod>2018-11-03</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/paint/</loc>
    <lastmod>2018-07-17</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/polar-clock/</loc>
    <lastmod>2022-06-24</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/polyrhythm/</loc>
    <lastmod>2024-12-23</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/quality-podcasts/</loc>
    <lastmod>2016-03-13</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/raining-light-effect/</loc>
    <lastmod>2022-11-06</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/raining-paint/</loc>
    <lastmod>2018-05-27</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/reverse-hover-state/</loc>
    <lastmod>2022-07-02</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/sound-frequency-slider/</loc>
    <lastmod>2023-10-16</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/sticky-stacky/</loc>
    <lastmod>2022-07-08</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/ten-star-movies/</loc>
    <lastmod>2016-11-15</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/unique-stars/</loc>
    <lastmod>2018-02-26</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/using-canvas-to-make-patterns-with-circles/</loc>
    <lastmod>2016-03-13</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/vertically-center-siblings/</loc>
    <lastmod>2021-04-29</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/wordle-solver-2/</loc>
    <lastmod>2023-12-30</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/wordle-solver/</loc>
    <lastmod>2022-04-22</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/wordscapes-solver/</loc>
    <lastmod>2022-12-23</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
  <url>
    <loc>https://briananders.com/posts/yahtzee/</loc>
    <lastmod>2020-12-17</lastmod>
    <priority>0.8</priority>
    <changefreq>yearly</changefreq>
  </url>
</urlset>